ToolBox/LeanDataWriter.cs

```diff
@@ -87,6 +100,229 @@ public void Write(IEnumerable<BaseData> source)
             }
         }
 
+        /// <summary>
+        /// Downloads historical data from the brokerage and saves it in LEAN format.
+        /// </summary>
+        /// <param name="brokerage">The brokerage from where to fetch the data</param>
+        /// <param name="symbols">The list of symbols</param>
+        /// <param name="startTimeUtc">The starting date/time (UTC)</param>
+        /// <param name="endTimeUtc">The ending date/time (UTC)</param>
+        public void DownloadAndSave(IBrokerage brokerage, List<Symbol> symbols, DateTime startTimeUtc, DateTime endTimeUtc)
+        {
+            if (symbols.Count == 0)
+            {
+                throw new Exception("The symbol list cannot be empty.");
+            }
+
+            if (_tickType != TickType.Trade && _tickType != TickType.Quote)
+            {
+                throw new Exception("The tick type must be Trade or Quote.");
+            }
+
+            var dataType = LeanData.GetDataType(_resolution, _tickType);
+
+            var marketHoursDatabase = MarketHoursDatabase.FromDataFolder();
+
+            var ticker = symbols.First().ID.Symbol;
+            var market = symbols.First().ID.Market;
+
+            var canonicalSymbol = Symbol.CreateFuture(ticker, market, SecurityIdentifier.DefaultDate);
+
+            var historyBySymbol = new Dictionary<Symbol, List<IGrouping<DateTime, BaseData>>>();
+            var historyBySymbolDailyOrHour = new Dictionary<Symbol, List<BaseData>>();
+
+            foreach (var symbol in symbols)
+            {
+                var exchangeHours = marketHoursDatabase.GetExchangeHours(symbol.ID.Market, symbol, _securityType);
+                var dataTimeZone = marketHoursDatabase.GetDataTimeZone(symbol.ID.Market, symbol, _securityType);
```

Comment:
`dataTimeZone` & `exchangeHours` should be shared by all contract symbols under the same Future?